datastore/key.go

```diff
@@ -18,6 +18,10 @@ import (
 
 	"google.golang.org/appengine/internal"
 	pb "google.golang.org/appengine/internal/datastore"
+
+	// Adding Support the new key encoding format
+	// in cloud.google.com/go/datastore
+	nds "cloud.google.com/go/datastore"
```

Comment:
I see your point.  Evaluating this, it seems to be about the same amount of code duplication that we need from google.golang.org/appengine/datastore.  I am concerned that by duplicating the code from the new to the old lib than an update to the new lib will break the functionality because the protoToKey() function will need to be duplicated.  My assumption is that developers will be leveraging both google.golang.org/appengine/datastore and cloud.google.com/go/datastore going forward and so will already have both libs.  I defer you your wisdom on this.  It will take me an extra day and a half of dev time to make this change if you consider it a must have.

Reply:
+1, could this live in another package, or within the Cloud client libraries?

Could we import (or copy) just the protobufs, rather than the whole library?

Reply:
Presumably the more common long-term case is that people will need to decode the older format (App Engine) and use it it with [new libraries](https://godoc.org/cloud.google.com/go/datastore#DecodeKey) (Cloud Datastore).

That would make me lean towards adding support for the old format to cloud.google.com/go/datastore. But you probably understand the use case better than me.

Reply:
I am working on the solution for old format.  So far, it is much easier than this solution because there is not a dependency on the appid.  As for this lib, duplicating the code from cloud.google.com/go/datastore is going to be a challenge.  They way protoToKey() works is different and we will have to duplicate code from internal and proto I believe.

Reply:
Complete